assemblyline/templates/ingress.yaml

```diff
@@ -41,38 +40,124 @@ spec:
                   number: 5002
+
+---
+
+# UI INGRESS (Internal)
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Release.Name }}-ui-ingress
+  annotations:
+    {{- range $key, $value := .Values.ingressAnnotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+    appgw.ingress.kubernetes.io/backend-hostname: "ui"
+spec:
+  tls:
+    - hosts: [{{ .Values.configuration.ui.fqdn }}]
+      secretName: {{ if .Values.tlsSecretName }} {{.Values.tlsSecretName }} {{else}} {{ .Release.Name }}-generated-cert {{end}}
+  rules:
+    - host: {{ if .Values.ingressHost }} {{.Values.ingressHost }} {{else}} {{ .Values.configuration.ui.fqdn | quote }} {{end}}
+      http:
+        paths:
           - path: /api/
             pathType: Prefix
             backend:
               service:
                 name: ui
                 port:
                   number: 5000
-          {{ if .Values.separateIngestAPI }}
+
+---
+
+# UI-INGEST INGRESS (Internal)
+{{ if .Values.separateIngestAPI }}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Release.Name }}-ui-ingest-ingress
+  annotations:
+    {{- range $key, $value := .Values.ingressAnnotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+    appgw.ingress.kubernetes.io/backend-hostname: "ui"
+spec:
+  tls:
+    - hosts: [{{ .Values.configuration.ui.fqdn }}]
+      secretName: {{ if .Values.tlsSecretName }} {{.Values.tlsSecretName }} {{else}} {{ .Release.Name }}-generated-cert {{end}}
+  rules:
+    - host: {{ if .Values.ingressHost }} {{.Values.ingressHost }} {{else}} {{ .Values.configuration.ui.fqdn | quote }} {{end}}
+      http:
+        paths:
           - path: /api/v4/ingest/
             pathType: Prefix
             backend:
               service:
                 name: ui-ingest
                 port:
                   number: 5000
-          {{end}}
-          {{ if .Values.internalELKStack}}
+{{end}}
+
+---
+
+# KIBANA INGRESS (Internal)
+{{ if .Values.internalELKStack}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Release.Name }}-kibana
+  annotations:
+    {{- range $key, $value := .Values.ingressAnnotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+    appgw.ingress.kubernetes.io/backend-hostname: "kibana"
+spec:
+  tls:
+    - hosts: [{{ .Values.configuration.ui.fqdn }}]
+      secretName: {{ if .Values.tlsSecretName }} {{.Values.tlsSecretName }} {{else}} {{ .Release.Name }}-generated-cert {{end}}
+  rules:
+    - host: {{ if .Values.ingressHost }} {{.Values.ingressHost }} {{else}} {{ .Values.configuration.ui.fqdn | quote }} {{end}}
+      http:
+        paths:
           - path: /kibana/
             pathType: Prefix
             backend:
               service:
                 name: kibana
                 port:
                   number: 5601
-          {{ end }}
+{{end}}
+
+---
 {{ if .Values.ingressPaths }}
 {{ .Values.ingressPaths | toYaml | indent 10 }}
-{{ end }}
+{{ end }}
```

Comment:
How do we want to handle this configuration for appending ingress paths to what was once a single Ingress definition?